File: 2022-2023/sourcecode/lab/lab9/RoomRelationsJava/app/src/main/java/pl/edu/uwr/pum/roomrelationsjava/model/entities/relations/StudentWithLectures.java
package pl.edu.uwr.pum.roomrelationsjava.model.entities.relations;

import androidx.room.Embedded;
import androidx.room.Junction;
import androidx.room.Relation;

import java.util.List;

import pl.edu.uwr.pum.roomrelationsjava.model.entities.Lecture;
import pl.edu.uwr.pum.roomrelationsjava.model.entities.Student;

public class StudentWithLectures {
    @Embedded
    public Student student;

    @Relation(
            parentColumn = "studentName",
            entityColumn = "lectureName",
            associateBy = @Junction(StudentLectureCrossRef.class)
    )

    public List<Lecture> lectureList;
}

File: 2022-2023/sourcecode/lab/lab9/RoomRelationsKotlin/app/src/main/java/pl/edu/uwr/pum/roomrelationskotlin/model/entity/relations/StudentWithLectures.kt
package pl.edu.uwr.pum.roomrelationskotlin.model.entity.relations

import androidx.room.Embedded
import androidx.room.Junction
import androidx.room.Relation
import pl.edu.uwr.pum.roomrelationskotlin.model.entity.Lecture
import pl.edu.uwr.pum.roomrelationskotlin.model.entity.Student

data class StudentWithLectures (
    @Embedded val student: Student,

    @Relation(
        parentColumn = "studentName",
        entityColumn = "lectureName",
        associateBy = Junction(StudentLectureCrossRef::class)
    )

    val lectures: List<Lecture>
    )